backend/controllers/taskController.js

```diff
@@ -15,4 +19,101 @@ const createTask = asyncHandler(async (req, res) => {
   }
 });
 
-module.exports = { createTask };
+const getTasks = asyncHandler(async (req, res) => {
+  // TESTED
+  // console.log("in GetTasks");
+  const userId = req.body.user_id;
+  console.log(userId);
+  const tasks = await Task.find({ user_id: userId });
+
+  // console.log("TASKS:");
+  // console.log(tasks);
+  res.status(200).json(tasks);
+});
+
+const getTasksById = asyncHandler(async (req, res) => {
+  // Task id Parameter
+  console.log("In getTasksById");
+  const id = req.params.id;
+
+  // Users id
+  const userId = req.body.user_id;
+
+  const tasks = await Task.find({
+    user_id: userId,
+  });
+  // .findById(req.body.id);
+  console.log(tasks);
+  console.log(id);
+  const userTasks = tasks.filter(function (task) {
```

Comment:
If I'm not mistaken, I think we can apply the filtering when searching for the tasks in `Task.find()` no? (Rather than having a separate filter that will loop through all the tasks for the specific one that we're looking for

Reply:
Additionally, since we're obtaining a task by id, only 1 task should be returned rather than multiple. (i.e. Your function should be `getTaskbyId` and following variables that reference `tasks` should just be singular as `task`

Reply:
Fixed: Changes are reflected in recent commits